src/main/java/mate/academy/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package mate.academy.service.impl;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import mate.academy.dao.OrderDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Injector;
+import mate.academy.lib.Service;
+import mate.academy.model.Order;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.User;
+import mate.academy.service.OrderService;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+    private static Injector injector = Injector.getInstance("mate.academy");
+    @Inject
+    private OrderDao orderDao;
+    private ShoppingCartService shoppingCartService = (ShoppingCartService) injector
+            .getInstance(ShoppingCartService.class);
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        Order order = new Order();
+        order.setOrderDate(LocalDateTime.now());
+        order.setUser(shoppingCart.getUser());
+        order.setTickets(new ArrayList(shoppingCart.getTickets()));
+        shoppingCartService.clearShoppingCart(shoppingCart);
```

Comment:
Ensure that the `completeOrder` method correctly handles the transfer of tickets from the `ShoppingCart` to the `Order`. According to the checklist, you should pass a copy of the tickets list to avoid Hibernate automatically deleting them from the `ShoppingCart` in the database. The use of `new ArrayList(shoppingCart.getTickets())` is correct for creating a copy of the list. Additionally, ensure that `shoppingCartService.clearShoppingCart(shoppingCart)` effectively clears the shopping cart after creating the order.